lib/Frontend/ModuleInterfaceSupport.cpp

```diff
@@ -351,7 +359,11 @@ static void printImports(raw_ostream &out,
         out << "@_spi(" << spiName << ") ";
     }
 
-    if (ctx.LangOpts.hasFeature(Feature::InternalImportsByDefault)) {
+    if (!Opts.printPublicInterface() &&
+        Opts.printPackageInterface() &&
```

Comment:
Could we remove these two checks to simplify the condition? If `packageOnlyImportSet.count()` returns something above 0 it means these checks passed previously.